kernel/byterun/coq_interp.c

```diff
@@ -925,10 +925,14 @@ value coq_interprete
         mlsize_t i;
         value block;
         print_instr("MAKEBLOCK, tag=");
-        Coq_alloc_small(block, wosize, tag);
-        Field(block, 0) = accu;
-        for (i = 1; i < wosize; i++) Field(block, i) = *sp++;
-        accu = block;
+        if (wosize == 0) {
+          accu = Atom(tag);
```

Comment:
Probably not worth it, but we know that the tag will always be 0 in the case of Coq so maybe we could just assert it.

Reply:
I have considered it, but that might send the wrong message, as it would look like a precondition of this code, while it works perfectly fine for non-zero tags.